src/shared-library.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2019 Jamie Madill
+ *
+ * This file is part of the glmark2 OpenGL (ES) 2.0 benchmark.
+ *
+ * glmark2 is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * glmark2 is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * glmark2.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *  Jamie Madill
+ */
+#include "shared-library.h"
+#include <dlfcn.h>
+#include <stdio.h>
+
+namespace
+{
+// Try both system and development  extensions on posix systems. Prioritize the
+// development extension for users who are doing driver development.
+const char *extensions[] = { ".so", ".so.1" };
```

Comment:
GLES2 has a library name of libGLESv2.so.**_2_** so we would also need ".so.2". Perhaps, though, a cleaner approach would be to allow the caller to provide a list of alternatives instead of trying to guess. Something like:

bool SharedLibrary::open_from_alternatives(std::initializer_list<const char*> alt_names);
lib.open_from_alternatives({"libGLESv2.so", "libGLESv2.so.2"});

Reply:
Done. This will require a bit different logic on Windows. But can deal with it when we get there.